src/llmq/quorums_blockprocessor.cpp

```diff
@@ -109,10 +110,22 @@ bool CQuorumBlockProcessor::ProcessBlock(const CBlock& block, const CBlockIndex*
 {
     AssertLockHeld(cs_main);
 
+    // Make sure evodb is consistent.
+    // If we already have best block hash saved, the previous block should match it.
+    // If we don't, it's ok too but only before DIP3 activation.
+    uint256 hashBestBlock = uint256();
+    bool fHasHashBestBlock = evoDb.Read(DB_BEST_BLOCK, hashBestBlock);
+    uint256 hashPrevBlock = fHasHashBestBlock ? pindex->pprev->GetBlockHash() : uint256();
```

Comment:
May be it is better to check that `pindex` and `pindex->pprev` are not empty before using them?